cmd/krew/cmd/list.go

```diff
@@ -24,11 +24,89 @@ import (
 
 	"github.com/pkg/errors"
 	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/cli-runtime/pkg/printers"
+	"k8s.io/kubectl/pkg/util/slice"
 
+	"sigs.k8s.io/krew/pkg/constants"
 	"sigs.k8s.io/krew/pkg/installation"
 )
 
+type Entry struct {
+	Name    string
+	Version string
+}
+
+type Entries []Entry
+
+func sortByName(e Entries) Entries {
+	sort.Slice(e, func(a, b int) bool {
+		return e[a].Name < e[b].Name
+	})
+	return e
+}
+
+// Consume produces a junk GroupVersionKind for obj.GetObjectKind().GroupVersionKind().Empty() check to eat in PrintObj()
+type Consume struct {
+	APIVersion string
+	Kind       string
+}
+
+func (c Consume) SetGroupVersionKind(kind schema.GroupVersionKind) {
+	c.APIVersion, c.Kind = kind.ToAPIVersionAndKind()
+}
+
+func (c Consume) GroupVersionKind() schema.GroupVersionKind {
+	return schema.FromAPIVersionAndKind(c.APIVersion, c.Kind)
+}
+
+func (e Entries) GetObjectKind() schema.ObjectKind {
+	return Consume{constants.CurrentAPIVersion, constants.PluginKind}
+}
+
+func (e Entries) DeepCopyObject() runtime.Object {
+	return append(Entries{}, e...)
+}
+
+type ListFlags struct {
+	JSONYamlPrintFlags *genericclioptions.JSONYamlPrintFlags
+	OutputFormat       *string
```

Comment:
`OutputFormat` doesn't need to be a pointer, right? What about
```patch
diff --git a/cmd/krew/cmd/list.go b/cmd/krew/cmd/list.go
index c478981..e592a94 100644
--- a/cmd/krew/cmd/list.go
+++ b/cmd/krew/cmd/list.go
@@ -72,15 +72,11 @@ func (e Entries) DeepCopyObject() runtime.Object {
 
 type ListFlags struct {
 	JSONYamlPrintFlags *genericclioptions.JSONYamlPrintFlags
-	OutputFormat       *string
+	OutputFormat       string
 }
 
 func NewListFlags() *ListFlags {
-	outputFormat := ""
-	return &ListFlags{
-		JSONYamlPrintFlags: genericclioptions.NewJSONYamlPrintFlags(),
-		OutputFormat:       &outputFormat,
-	}
+	return &ListFlags{JSONYamlPrintFlags: genericclioptions.NewJSONYamlPrintFlags()}
 }
 
 func (f *ListFlags) AllowedFormats() []string {
@@ -88,23 +84,21 @@ func (f *ListFlags) AllowedFormats() []string {
 }
 
 func (f *ListFlags) ToPrinter() (printers.ResourcePrinter, error) {
-	outputFormat := *f.OutputFormat
 	var printer printers.ResourcePrinter
 
-	switch outputFormat {
+	switch f.OutputFormat {
 	case "json":
 		printer = &printers.JSONPrinter{}
 	case "yaml":
 		printer = &printers.YAMLPrinter{}
 	default:
-		return nil, genericclioptions.NoCompatiblePrinterError{OutputFormat: f.OutputFormat, AllowedFormats: f.AllowedFormats()}
+		return nil, genericclioptions.NoCompatiblePrinterError{OutputFormat: &f.OutputFormat, AllowedFormats: f.AllowedFormats()}
 	}
 
 	return printer, nil
 }
 
 func init() {
-	outputFormat := ""
 	output := NewListFlags()
 
 	// listCmd represents the list command
@@ -124,7 +118,7 @@ Remarks:
 			}
 
 			// return sorted list of plugin names when piped to other commands or file
-			if *output.OutputFormat == "name" || (*output.OutputFormat == "" && !isTerminal(os.Stdout)) {
+			if output.OutputFormat == "name" || (output.OutputFormat == "" && !isTerminal(os.Stdout)) {
 				var names []string
 				for name := range plugins {
 					names = append(names, name)
@@ -134,7 +128,7 @@ Remarks:
 				return nil
 			}
 
-			if *output.OutputFormat == "wide" || (*output.OutputFormat == "" && isTerminal(os.Stdout)) {
+			if output.OutputFormat == "wide" || (output.OutputFormat == "" && isTerminal(os.Stdout)) {
 				// print table
 				var rows [][]string
 				for p, version := range plugins {
@@ -144,7 +138,7 @@ Remarks:
 				return printTable(os.Stdout, []string{"PLUGIN", "VERSION"}, rows)
 			}
 
-			if slice.ContainsString(output.AllowedFormats(), *output.OutputFormat, nil) {
+			if slice.ContainsString(output.AllowedFormats(), output.OutputFormat, nil) {
 				objs := make(Entries, 0, len(plugins))
 				for plugin, version := range plugins {
 					obj := Entry{plugin, version}
@@ -167,9 +161,8 @@ Remarks:
 		PreRunE: checkIndex,
 	}
 
-	listCmd.Flags().StringVarP(&outputFormat, "output", "o", "", "Output format. One of json, yaml, wide, name")
+	listCmd.Flags().StringVarP(&output.OutputFormat, "output", "o", "", "Output format. One of json, yaml, wide, name")
 
-	output.OutputFormat = &outputFormat
 	rootCmd.AddCommand(listCmd)
 }
 
```

Reply:
I kept OutputFormat a pointer for consistency with genericclioptions. This switch looks good to me, commit it?